src/System.Private.CoreLib/shared/System/Globalization/IdnMapping.cs

```diff
@@ -707,7 +707,7 @@ private static string PunycodeDecode(string ascii)
                         i %= (output.Length - iOutputAfterLastDot - numSurrogatePairs + 1);
 
                         // Make sure n is legal
-                        if (n < 0 || n > 0x10ffff || (n >= 0xD800 && n <= 0xDFFF))
+                        if ((uint)n > 0x10ffff || (n >= 0xD800 && n <= 0xDFFF))
```

Comment:
Hooray, a simplification! 😀
`if (!Rune.IsValid(n))`

Perhaps out of scope here, but there's likely some low-hanging fruit throughout the Framework that could be touched up in this manner.